### DISCUSSION TITLE
Importing CAD data with named selections from Discovery

### DISCUSSION THREAD

**Discussion by aglowinski-danfoss** [ansys/pyprimemesh#914] [Q&A]
Dear Developers,
the Ansys support told us, to switch from SpaceClaim to Discovery, because SC will not get any updates in the near future.
Therefore, we're now trying to import dsco CAD file into PyPrime. 
If I understand this entry correctly (https://prime.docs.pyansys.com/version/stable/api/_autosummary/ansys.meshing.prime.FileIO.import_cad.html), DSCO file are not supported yet? 
Is there any other file format I can export from Discovery, that contains the CAD as well as the named selection data and can be imported in PyPrime?
Best regards,
Alex

**Comment by waltersma** [comment]
Hi Alex,

This is actually an issue with the docstring that we will fix shortly.  Thanks for spotting this.  DSCO files are indeed supported for windows.  The files will be imported using an out of process instance of Discovery so you will need this installed and configured to import the files.  

Thanks

**Reply by aglowinski-danfoss** [reply]
Hi @waltersma,
unfortunately I am still failing to import dsco files. Discovery is installed and used for creating the files for PyPrime, so I guess this is not the problem.
Normally I would use:
mesh_util.read(file_name="dsco_file.dsco", cad_reader_route=prime.CadReaderRoute.DISCOVERY)
But this does not work for me. Am I missing something? Is my prime client not up to date?
BR Alex

**Reply by waltersma** [reply]
Hi Alex, 

What error do you see?  This looks ok to me.  What release of Ansys and PyPrimeMesh client are you using?  You can also export FMD format from Discovery and this will preserve the named selections you have defined.  Can you try reading one of the examples to see if it is something specific to the file you are reading?

For example:
```
import ansys.meshing.prime.graphics as graphics
from ansys.meshing import prime
from ansys.meshing.prime import lucid

prime_client = prime.launch_prime()

model = prime_client.model
mesh_util = lucid.Mesh(model)

pipe_tee = prime.examples.download_pipe_tee_dsco()
mesh_util.read(pipe_tee)

display = graphics.Graphics(model)
display()
prime_client.exit()
```
